src/util.coffee

```diff
@@ -32,3 +32,25 @@
                                       node?.tag is 'tag:yaml.org,2002:map'
 
 @NON_PRINTABLE = /[^\x09\x0A\x0D\x20-\x7E\x85\xA0-\uD7FF\uE000-\uFFFD]/
+
+@stringifyJSONCircularDepth = (json, depth = 1000) ->
```

Comment:
What I meant is, why have an arbitrary length at all, the first time you detect a cycle, you should abort (just like we do in other parts)